specs/unified_kernel_image.md

```diff
@@ -159,3 +159,22 @@ that tools staging or consuming UKIs have a common place to store and look for t
 
 The installed UKIs should have a filename `<version format specification>.efi`, i.e. the filename is left to
 implementers but must be valid for comparisons according to the [Version Format Specification](version_format_specification.md).
+
+UKI addons (cmdline PE addons, systemd-sysext extensions for the initrd) built centrally by distributions and
+installed via package manager should be installed into locations depending on whether they are global UKI addons,
+that should be applied to all UKIs installed in the ESP, or whether they should only be applied to a
+single specific UKI.
+
+Global UKI addons should be installed into `/usr/lib/modules/$UNAME/extra.d/`. UKI-specific addons should be installed
+to `/usr/lib/modules/$UNAME/$UKI.efi.extra.d/`, where `$UNAME` is the output of `uname -r` of the kernel included
+in the UKI and `$UKI` is the name of the corresponding centrally built UKI with the `.efi` extension stripped.
```

Comment:
Hmm. The first part of the text says "global UKI addons, that should be applied to all UKIs installed in the ESP". When I read that, I expect a path that doesn't refer to `$UNAME`. I would expect a path like `/usr/lib/modules/extra.d`. This would apply to any addon which doesn't depend strongly on the kernel version.

Reply:
Keeping in mind that anything is fine for me at this point, I also initially thought of having global add-ons in modules/extra.d/. But then think about multiple rpms of multiple kernel version, each installing its own global addons. Maybe for kernel X the global addon test.addon.efi has test=1, for kernel Y the same addon has test=2?
Does it make sense to you? Or would that not be a problem?

Reply:
Well, if you have multiple rpms that want to provide global addons with the same name, that clearly wouldn't work. A global addon would need to be packaged separately.